protocol/src/protocol.selectionRange.md

```diff
@@ -0,0 +1,16 @@
+#### Selection Range Request (:leftwards_arrow_with_hook:)
+
+The selection range request is sent from the client to the server to return suggested selection ranges at a given position.
+The first range must contain the given position.
+Position can coincide with the start or end of the first range.
+Subsequent ranges must contain the previous range.
```

Comment:
I think more documentation is required to clarify what this request is supposed to do. I personally find the term "selection range" a little vague but maybe that's just me.